pkg/generate/git/repository.go

```diff
@@ -337,6 +348,42 @@ func command(w io.Writer, name, dir string, env []string, args ...string) (stdou
 	return
 }
 
+// timedCommand executes an external command in the given directory with a timeout.
+// The command's standard out and error are trimmed and returned as strings
+// It may return the type *GitError if the command itself fails.
+func timedCommand(timeout time.Duration, w io.Writer, name, dir string, env []string, args ...string) (stdout, stderr string, err error) {
```

Comment:
I don't see how making every single consumer of this method, sans one, pass a magic value to signal that we don't want a timeout is better than what I've written.